src/main/java/net/javadiscord/javabot/systems/help/AutoCodeFormatter.java

```diff
@@ -0,0 +1,164 @@
+package net.javadiscord.javabot.systems.help;
+
+import lombok.RequiredArgsConstructor;
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+import net.javadiscord.javabot.data.config.BotConfig;
+import net.javadiscord.javabot.systems.moderation.AutoMod;
+import net.javadiscord.javabot.systems.user_preferences.UserPreferenceService;
+import net.javadiscord.javabot.systems.user_preferences.model.Preference;
+import net.javadiscord.javabot.util.ExceptionLogger;
+import net.javadiscord.javabot.util.WebhookUtil;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.Nonnull;
+import java.util.Objects;
+
+/**
+ * Handles auto-formatting of code in help-channels.
+ * A non-listener class since it has to be called strictly after {@link HelpListener} to not send its messages before channel reserving.
+ */
+@RequiredArgsConstructor
+@Component
+public class AutoCodeFormatter {
+	private final AutoMod autoMod;
+	private final BotConfig botConfig;
+	private final UserPreferenceService preferenceService;
+
+	/**
+	 * Method responsible for finding a place to insert a codeblock, if present.
+	 *
+	 * @param event a {@link MessageReceivedEvent}.
+	 * @return a MessageCodeblock instance, holding a startIndex, content and
+	 * an endIndex. Returns null if no place was found.
+	 */
+	@Nullable
+	private static CodeBlock findCodeblock(@NotNull MessageReceivedEvent event) {
```

Comment:
For discussion: Should messages with the first `{` ad the last `}` in the same line (e.g. `/guilds/{id}`) be considered?
If so, should we maybe generally exclude all messages that even have a single backtick and optimistically assume they know how to format code?
Or should there maybe be a more complex check that e.g. ignores lines with the same amount of opening and closing brackets (I don't think that's necessary though)?

Reply:
I'd say we just try it this way, and optimize if needed?